server/web_ui/src/views/identityverification.rs

```diff
@@ -0,0 +1,533 @@
+                <form
+                onsubmit={ ctx.link().callback_future(move |e: SubmitEvent| {
+                    #[cfg(debug_assertions)]
+                    console::debug!("identity-verification::view_state -> Init - prevent_default()".to_string());
+                    e.prevent_default();
+                    self_clone.to_owned().get_transition_from_submit_code()
+                } ) }
+                >
+                <div class={CLASS_DIV_LOGIN_FIELD}>
+                    <input
+                        autofocus=true
+                        class="autofocus form-control"
+                        id="totp_code"
+                        name="code"
+                        type="text"
+                        autocomplete="code"
+                        oninput={Self::check_input_callback(ctx, totp.to_string())}
+                        value={ totp.to_string() }
+                    />
+                </div>
+
+                <div class={CLASS_DIV_LOGIN_BUTTON}>
+                    <button
+                        type="submit"
+                        class="btn btn-primary"
+                        disabled={ !totp_valid }
+                    >{" Continue "}</button>
+                </div>
+                </form>
+            </div>
+            </div>
+        }
+    }
+
+    fn view_error(&self, error_message: &str) -> Html {
+        html! {
+          <>
+            <p class="text-center">
+                <img src="/pkg/img/logo-square.svg" alt="Kanidm" class="kanidm_logo"/>
+            </p>
+            <div class="alert alert-danger" role="alert">
+              <h2>{ "An Error Occurred 🥺" }</h2>
+            <p>{ error_message }</p>
+            </div>
+            <p class="text-center">
+              <a href="/"><button href="/" class="btn btn-secondary" aria-label="Return home">{"Return to the home page"}</button></a>
+            </p>
+          </>
+        }
+    }
+
+    fn view_success(&self) -> Html {
+        let other_id = self.other_id.clone().unwrap_or_default();
+        html! {
+          <>
+            <div class="alert alert-success" role="alert">
+            <h4 class="alert-heading">{"Success 🎉🎉"}</h4>
+            <p><b>{other_id}</b>{"'s identity has been successfully confirmed!"}</p>
+            </div>
+            <p class="text-center">
+              <a href="/"><button href="/" class="btn btn-secondary" aria-label="Return home">{"Return to the home page"}</button></a>
+            </p>
+          </>
+        }
+    }
+
+    // the purpose of the following functions is to get what to do next in the state machine.
+    // each main view has its own function, that is the start view (even tho it's displayed for few ms), the id_display_and_submit view
+    // and the submit_code
+
+    async fn get_transition_from_start(state: IdentifyUserState) -> IdentifyUserTransition {
+        let request = match state {
```

Comment:
This might be better as a debug assert since the only time this is called is during create?